### PR TITLE
Manually merge 3.0.x into master

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: src/Platforms/AbstractPlatform.php [hunk 1]
```diff
@@ -731,18 +731,21 @@ public function getNowExpression(): string
...
+     * SQLite only supports the 2 parameter variant of this function.
+     *
+     * @param string   $string An sql string literal or column name/alias.
+     * @param int      $start  Where to start the substring portion.
+     * @param int|null $length The substring portion length.
+     *
+     * @return string
      */
-    public function getSubstringExpression(string $string, string $start, ?string $length = null): string
+    public function getSubstringExpression($string, $start, $length = null)
```
Comment (greg0ire) [inline src/Platforms/AbstractPlatform.php:742]: Ah I think this is where all the issues come from, isn't it? :sweat_smile: bad merge resolution.

Comment (greg0ire) [reply src/Platforms/AbstractPlatform.php]: We don't, that's why I say "bad merge resolution", and why this diff is marked as outdated :smile: sorry for the confusion.

Comment (morozov) [reply src/Platforms/AbstractPlatform.php]: Sorry, I didn't notice the Outdated pill.

Comment (greg0ire) [reply src/Platforms/AbstractPlatform.php]: No worries :)

**Review by morozov — APPROVED** [review]

**Pull request merged**